docs/SERVER-API.md

```diff
@@ -0,0 +1,214 @@
+
+**Response:** Finished opening new pane - indicates that the server has finished closing the pane. This exists for the purpose of synchronization @@@what's blocked?
+
+### Switch focus pane
+
+**Purpose:** Switch the focus [pane](./TERMINOLOGY.md#pane). The focus pane is the target for many other pane-related actions.
+
+**Parameters:** @@@none?
+
+**Response:** None.
+
+### Move focus pane
+
+**Purpose:**
+
+**Parameters:** 
+
+**Response:** 
+
+### Toggle fullscreen focus pane
+
+**Purpose:** In a normal layout, switches the focus [pane](./TERMINOLOGY.md#pane) to be fullscreen. When the focus pane is fullscreen, switches back to the normal layout, potentially with multiple panes.
+
+**Parameters:** None.
+
+**Response:** Whether the layout has the focus pane locked to fullscreen or not.
+
+### Resize focus pane
+
+**Purpose:** Resizes the focus [pane](./TERMINOLOGY.md#pane). 
+
+**Parameters:** Direction in which the pane should be resized. @@@ When does it shrink, when does it grow? Is it the top left that's fixed?
+
+**Response:** None.
+
+### Horizontal split
+
+**Purpose:** Split the current focus [pane](./TERMINOLOGY.md#pane) horizontally, creating a new pane @@@above or below?
+
+**Parameters:** None.
+
+**Response:** Done opening new pane - indicates that the server has finished opening the pane. This exists for the purpose of synchronization @@@what's blocked?
+
+### Vertical split
+
+**Purpose:** Split the current focus [pane](./TERMINOLOGY.md#pane) vertically, creating a new pane to the @@@left or right?
+
+**Parameters:** None.
+
+**Response:** Done opening new pane - indicates that the server has finished opening the pane. This exists for the purpose of synchronization @@@what's blocked?
+
+
+## Tab controls
+
+### New tab
+
+**Purpose:** Opens a new [tab](./TERMINOLOGY.md#tab).
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Close tab
+
+**Purpose:** Closes the current [tab](./TERMINOLOGY.md#tab).
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Next tab
+
+**Purpose:** Switches to the next [tab](./TERMINOLOGY.md#tab) by index.
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Previous tab
+
+**Purpose:** Switches to the previous [tab](./TERMINOLOGY.md#tab) by index.
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Go to tab
+
+**Purpose:** Switches to a specific [tab](./TERMINOLOGY.md#tab).
+
+**Parameters:** Index - the index of the tab within the list. @@@ Shouldn't this really be based on name? Index is quite internal.
+
+**Response:** None.
+
+### Last used tab
+
+**Purpose:** Switches to the last [tab](./TERMINOLOGY.md#tab) that was in use.
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Rename tab
+
+**Purpose:** Renames the current [tab](./TERMINOLOGY.md#tab).
+
+**Parameters:** A vector of bytes containing the new name for the tab.
+
+**Response:** None.
+
+## Session management
+
+### New client
+
+**Purpose:** Indicates that a new client is connecting to the server.
+
+**Parameters:**
```

Comment:
As described above, I think we could do with merging the open file API into this.